### PR TITLE
appconfg api

### PR DESCRIPTION
**Pull request by ArtificialOwl**
Documentation related to the new AppConfig API

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

**Review by susnux — APPROVED** [review]

File: admin_manual/configuration_server/occ_command.rst [hunk 1]
```diff
@@ -341,6 +341,21 @@ The ``config`` commands are used to configure the Nextcloud server::
...
+     - ``--value=VALUE`` change the configuration value
+     - ``--type=TYPE`` change the type of the value. Use carefully; can break your instance
+     - ``--lazy|--no-lazy`` set value as `lazy`
+     - ``--sensitive|--no-sensitive`` set value as `sensitive`
+     - ``--update-only`` only updates if a value is already stored
+
+.. note::
+	See `Appconfig Concepts`_ to learn more about `typed value`, `lazy` and `sensitive` flag.
+
+.. _Appconfig Concepts: https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/config/appconfig.html#concept-overview
```
Comment (ChristophWurst) [inline admin_manual/configuration_server/occ_command.rst:356]: This should be a relative link.

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.rst [hunk 2]
```diff
@@ -29,7 +29,7 @@ Added APIs
 Changed APIs
 ^^^^^^^^^^^^
 
-* tbd
+* `IAppConfig` has been fully reworked, most of previous methods are now deprecated. The new version of the API implements multiple settings for app config values to define their laziness and sensitivity.
```
Comment (ChristophWurst) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.rst:32]: Link to developer_manual/digging_deeper/config/appconfig.rst missing

File: developer_manual/digging_deeper/config/appconfig.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
+=========
+AppConfig
```
Comment (ChristophWurst) [inline developer_manual/digging_deeper/config/appconfig.rst:2]: Shouldn't this go into https://github.com/nextcloud/documentation/blob/845fb9a79fe360b762690e735fcab236f9c16898/developer_manual/basics/storage/configuration.rst? There are now two pages for configuration

File: admin_manual/configuration_server/occ_command.rst (hunk 1 above)
Comment (ArtificialOwl) [reply admin_manual/configuration_server/occ_command.rst:356]: All link are made the same way, cannot find an example follow.
Should I just replace `https://docs.nextcloud.com/server[...]` by `/server[...]`

File: developer_manual/digging_deeper/config/appconfig.rst (hunk 3 above)
Comment (ChristophWurst) [reply developer_manual/digging_deeper/config/appconfig.rst:2]: Yes, migrate the new content into the old page

File: admin_manual/configuration_server/occ_command.rst (hunk 1 above)
Comment (ChristophWurst) [reply admin_manual/configuration_server/occ_command.rst:356]: I didn't realize that it's a link from admin docs to dev docs. You can leave it :+1:

**Pull request merged**